File: Adapter/python/PrinterAdapter.py
from Printer import Printer
from LegacyPrinter import LegacyPrinter

class PrinterAdapter(Printer):
    legacy_printer: LegacyPrinter
    
    def __init__(self, legacy_printer: LegacyPrinter):
        self.legacy_printer = legacy_printer
        
    def print(self, message: str):
        self.legacy_printer.printDocument(message)

File: Adapter/ts/PrinterAdapter.ts
import { Printer } from './Printer';
import { LegacyPrinter } from './LegacyPrinter';

export class PrinterAdapter implements Printer {
    legacy_printer: LegacyPrinter

    constructor(legacy_printer: LegacyPrinter) {
        this.legacy_printer = legacy_printer
    }

    print(message: string) {
        this.legacy_printer.printDocument(message)
    }
}
